src/calendar/calendar-month.tsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from "react";
+import dayjs from "dayjs";
+import { MonthCell, MonthPickerContainer } from "./calendar-month.style";
+import { CalendarMonthProps, VariantMonth } from "./types";
+
+export type Month = typeof Months[number];
+
+export const CalendarMonth = ({
+    calendarDate,
+    onClick,
+    showView,
+    selectedStartDate,
+}: CalendarMonthProps) => {
+    const [monthDate, setMonthDate] = useState<string[]>([""]);
+
+    useEffect(() => {
+        if (showView === "Month") {
+            generateMonth();
+        }
+    }, [showView, calendarDate]);
+
+    const generateMonthStatus = (date: string) => {
+        const arrayDate = date.split("-");
+        const month = arrayDate[1] as Month;
+        const value = dayjs(date).format("YYYY-MM-DD");
+        let variant: VariantMonth = "default";
+
+        if (!dayjs(value).isValid())
+            return {
+                month,
+                value,
+                variant,
+            };
+
+        variant = dayjs(selectedStartDate).isSame(value, "month")
+            ? "selectedMonth"
+            : dayjs().isSame(value, "month")
+            ? "currentMonth"
+            : "default";
+
+        return {
+            month,
+            value,
+            variant,
+        };
+    };
+
+    const generateMonth = () => {
```

Comment:
Thank you for this one. 

but it return dayjs object from this  `dayjs("2023-10-31").month(2)` instead of the number of the month.

Other,  return **31** day from `dayjs("2023-10-31").month(2).daysInMonth()`;

Reply:
sorry my example should have been `dayjs("2023-10-31").month(1)` (starts counting from 0), then `daysInMonth()` will return 28 for February

anyway I just wanted to show how to generate 1 dayjs object for every month, then they can be transformed to whatever format you need afterwards

Reply:
you so awesome. Thanks sharing that. 

I do a stupid way. hahah

Reply:
Please check this. I guess i did a stupid way again.  

`const [yyyy, , dd] = calendarDate.format("YYYY-MM-DD").split("-");`, to know what's the today day.

```
  const generateMonth = () => {
        const months: Dayjs[] = [];
        const [yyyy, , dd] = calendarDate.format("YYYY-MM-DD").split("-");

        for (let i = 0; i < 12; i++) {
            let month: Dayjs;
            month = dayjs(calendarDate).month(i);
            const dayInMonth: number = month.daysInMonth();

            if (+dd > dayInMonth) {
                // get last day of the month
                month = dayjs(`${yyyy}-${Months[i]}-${dayInMonth}`);
            }

            months.push(dayjs(month));
        }

        setMonthDate(months);
    };
```

Reply:
Let me know what needed. It look like didn't simplified it. 

I needed `const [yyyy, , dd] = calendarDate.format("YYYY-MM-DD").split("-")` because of one scenario as

User select Feb from 2023-01-31, that line of code is track the selected day or today day.



```
export const generateMonth = (calendarDate: Dayjs) => {
        const months: Dayjs[] = [];

        const [yyyy, , dd] = calendarDate.format("YYYY-MM-DD").split("-");

        for (let i = 0; i < 12; i++) {
            let month: Dayjs;
            month = dayjs(calendarDate).month(i);
            const dayInMonth: number = month.daysInMonth();

            if (+dd > dayInMonth) {
                // get last day of the month
                const mmmm = month.format("MMMM");
                month = dayjs(`${yyyy}-${mmmm}-${dayInMonth}`);
            }

            months.push(dayjs(month));
        }

        return months;
    };
```

Reply:
this is what I had in mind (pseudo code), no need to handle months with less days since dayjs adjusts it automatically
```
for (i = 0 to 11) {
  monthForSelectedDay = calendarDate.month(i)
  months.push(monthForSelectedDay)
}
```

btw should be `generateMonths` since it returns an array

Reply:
Thanks again for your time to review it again. 

The logic been simplified a lot. thanks!